specification/securityinsights/resource-manager/Microsoft.SecurityInsights/stable/2024-09-01/DataConnectors.json

```diff
@@ -0,0 +1,1696 @@
+          "description": "GCP Project Number"
+        },
+        "workloadIdentityProviderId": {
+          "type": "string",
+          "description": "GCP Workload Identity Provider ID"
+        }
+      },
+      "required": [
+        "serviceAccountEmail",
+        "projectNumber",
+        "workloadIdentityProviderId"
+      ],
+      "x-ms-discriminator-value": "GCP"
+    },
+    "GenericBlobSbsAuthModel": {
+      "description": "Model for API authentication for working with service bus or storage account.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/CcpAuthConfig"
+        }
+      ],
+      "type": "object",
+      "properties": {
+        "credentialsConfig": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Credentials for service bus namespace, keyvault uri for access key"
+        },
+        "storageAccountCredentialsConfig": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Credentials for storage account, keyvault uri for access key"
+        }
+      },
+      "x-ms-discriminator-value": "ServiceBus"
+    },
+    "GitHubAuthModel": {
+      "description": "Model for API authentication for GitHub. For this authentication first we need to approve the Router app (Microsoft Security DevOps) to access the GitHub account, Then we only need the InstallationId to get the access token from https://api.github.com/app/installations/{installId}/access_tokens.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/CcpAuthConfig"
+        }
+      ],
+      "type": "object",
+      "properties": {
+        "installationId": {
+          "type": "string",
+          "description": "The GitHubApp auth installation id."
+        }
+      },
+      "x-ms-discriminator-value": "GitHub"
+    },
+    "NoneAuthModel": {
+      "description": "Model for API authentication with no authentication method - public API.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/CcpAuthConfig"
+        }
+      ],
+      "type": "object",
+      "x-ms-discriminator-value": "None"
+    },
+    "JwtAuthModel": {
+      "description": "Model for API authentication with JWT. Simple exchange between user name + password to access token.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/CcpAuthConfig"
+        }
+      ],
+      "properties": {
+        "tokenEndpoint": {
+          "type": "string",
+          "description": "Token endpoint to request JWT"
+        },
+        "userName": {
+          "description": "The user name. If user name and password sent in header request we only need to populate the `value` property with the user name (Same as basic auth). If user name and password sent in body request we need to specify the `Key` and `Value`.",
+          "type": "object",
+          "additionalProperties": {
```

Comment:
>additionalProperties [](http://example.com/codeflow?start=11&length=20)

ARM is trying to reduce instances of unschematized APIs to make it easier for consumers of those APIs to understand them and use them reliably, not to mention provide better support in the various SDKs.

Please make this with a schema or provide an explanation for why you need this.

Reply:
@ razvanbadea-msft  we did the review on this property on the Preview version (I joined a few ARM meeting to get approval).